app/assets/javascripts/compute_resources/xenserver/cache_refresh.js

```diff
@@ -0,0 +1,20 @@
+function refreshCache(attribute_name, compute_resource_id, on_success) {
+    tfm.tools.showSpinner();
+    url = "/foreman_xen/cache/refresh";
```

Comment:
@HeWhoWas : I'd put the data-attribute on every select fields and pass the field to your function`refreshCache(this, ...)`. You can then access the data like this: `var url = $(item).data('url');`